src/Microsoft.AspNetCore.Mvc.Razor/Internal/MvcRazorDiagnosticSourceExtensions.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using Microsoft.AspNetCore.Mvc.Rendering;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Internal
+{
+    public static class MvcRazorDiagnosticSourceExtensions
+    {
+        public static void BeforeRazorView(
+            this DiagnosticSource diagnosticSource,
+            IRazorPage page,
+            ViewContext viewContext)
+        {
+            if (diagnosticSource.IsEnabled("Microsoft.AspNetCore.Mvc.Razor.BeforeRazorView"))
+            {
+                diagnosticSource.Write(
+                    "Microsoft.AspNetCore.Mvc.Razor.BeforeRazorView",
+                    new { page = page, viewContext = viewContext, });
```

Comment:
> No, the viewcontext gets you access to the two.

> yes it's redundant and we do that on purpose

# YES THESE NEED TO INCLUDE THE ACTION DESCRIPTOR AND HTTP CONTEXT.

We do that on purpose so that you don't have to maintain a million proxies.

If another event is missing those fields, they should be added.